src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/General/MaintenanceType.php

```diff
@@ -51,6 +76,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 SwitchType::class,
                 [
                     'required' => true,
+                    'attr' => [
+                        'multistore_configuration_key' => 'PS_SHOP_ENABLE',
+                        'disabled' => !$this->isAllShopContext && !$this->shopConfiguration->isOverridenByCurrentContext('PS_SHOP_ENABLE'),
```

Comment:
Of course this is due to previous version of admin configuration pages but if we used real configuration keys (here `PS_SHOP_ENABLE`  instead of `enable_shop`) we could do this without producing extra two lines for each field :|

Reply:
yep... I suggested it already on slack, it would definitely be nice, but it would be a breaking change...  and not everyone likes the idea of convention over configuration

Reply:
Actually the `multistore_configuration_key` attribute is also used to determine if we want the checkbox or not on this field, so this line would exist anyway.